src/plugins/ucx/ucx_backend.h

```diff
@@ -105,22 +105,12 @@ using nixlUcxCudaDevicePrimaryCtxPtr = std::shared_ptr<nixlUcxCudaDevicePrimaryC
 
 class nixlUcxEngine
     : public nixlBackendEngine {
-    private:
+    protected:
```

Comment:
It doesn't look like an optimal solution. Why do we need to give inheritors access to all private class members?